[BD-13] refactor: Clean up HTMLSnippet

Cup0fCoffee opened the pull request:
<!--

🫒🫒
🫒🫒🫒🫒         🫒 Note: the Olive master branch has been created.  Please consider whether your change
    🫒🫒🫒🫒     should also be applied to Olive. If so, make another pull request against the
🫒🫒🫒🫒         open-release/olive.master branch, or ping @nedbat for help or questions.
🫒🫒

🌰🌰🌰🌰🌰🌰     🌰 Note: the Nutmeg release is still supported.
                  Please consider whether your change should be applied to Nutmeg as well.

Please give your pull request a short but descriptive title.
Use conventional commits to separate and summarize commits logically:
https://open-edx-proposals.readthedocs.io/en/latest/oep-0051-bp-conventional-commits.html

Use this template as a guide. Omit sections that don't apply. You may link to information rather than copy it.
More details about the template are at https://github.com/openedx/open-edx-proposals/pull/180
(link will be updated when that document merges)
-->

## Description

This PR removes code from HTMLSnippets that is no longer needed after the conversion of all the XModules to standard XBlocks.

## Testing instructions

N/A

## Other information

To determine which code can be removed, I checked the commit history and grep'ed through the code. For the code that I removed, I wasn't able to find any places where it was being used, and the commit history showed that the places where it has been used before were removed. E.g. `git log -Gget_javascript -p`.

The remaining code of the `HTMLSnippet` class is being used in `xmodule/static_content.py`, which is used to collect javascript and css for XModules (for example when you run `paver update_assets`).

`Private-ref`: [BB-6731](https://tasks.opencraft.com/browse/BB-6731)

Cup0fCoffee committed:
refactor: clean up HTMLSnippet

Remove code that is no longer needed after conversion of all the
XModules to standard XBlocks.

openedx-webhooks commented:
Thanks for the pull request, @Cup0fCoffee!


When this pull request is ready, tag your edX technical lead.


<!-- comment:welcome-blended -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

Agrendalath commented on xmodule/x_module.py:
```diff
@@ -213,40 +213,12 @@ class HTMLSnippet:
     """
 
-    js = {}
-    js_module_name = None
```
Is it possible that this will raise an `AttributeError` [here](https://github.com/openedx/edx-platform/blob/a389a9ff10fd06bd053107fc6a4f19c7d24f7233/xmodule/mako_module.py#L71-L71)?

Cup0fCoffee replied:
What do you think about adding `js_module_name = None` to the `MakoTemplateBlockBase` then, since it's the only class the contains the method that uses the attribute?

Cup0fCoffee replied:
Added it in https://github.com/openedx/edx-platform/pull/31159/commits/0919fc7dabede63137749959dc84eb6d3c70a86f.

Agrendalath replied:
@Cup0fCoffee, this would be nice, but let's make sure that there are no classes that override `js_module_name` **after** a class inherits from `MakoTemplateBlockBase` and **before** it inherits from `HTMLSnippet` now. If it would, then changing this would override this attribute, as the value set in `MakoTemplateBlockBase` would take precedence in the MRO.

There are a few classes than inherit from both `MakoTemplateBlockBase` and `HTMLSnippet` - e.g. `SequenceBlock`, `RandomizeBlock`, `ConditionalBlock`. In their definitions, I can only see `XmlMixin` and `XModuleToXBlockMixin` between these two classes, but we should each of them to make sure

Cup0fCoffee replied:
Oh. I've never had to deal with MRO before, so it didn't even cross my mind to check. Here is the list of all the classes that inherit from both:
```
RandomizeBlock
LibraryContentBlock
ConditionalBlock
LTIBlock
PollBlock
SplitTestBlock
SequenceBlock
```

All of them have only `XmlMixin` and/or `XModuleToXBlockMixin` inbetween `MakoTemplateBlockBase` and `HTMLSnippet`.

Cup0fCoffee committed:
refactor: add js_module_name attribute to MakoTemplateBlockBase

In the previous commit `js_module_name = None` was removed from
HTMLSnippet class, because it's not being used by this class, or the
classes that inherit from it. But `js_module_name` is used in
MakoTemplateBlockBase.

Not all classes that inherit from MakoTemplateBlockBase define this
attribute, since the same classes inherit from HTMLSnippet class, which
used to provide a fallback for this attribute. To prevent
`AttributeError`, defining the default value of `None` in
MakoTemplateBlockBase.

Agrendalath reviewed (approved):
:+1:

* [x] I tested this: checked that XBlocks inheriting from `HTMLSnippet` work correctly
* [x] I read through the code
* [x] I checked for accessibility issues: n/a
* [x] Includes documentation: n/a
* [x] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository: n/a

Agrendalath commented:
@ormsbee, would you like to take a look at this before we merge?

ormsbee commented:
@Agrendalath: No need for me to review. You already flagged the one concern I would have (the wacky inheritance tree that these things are in). Please just coordinate with @jristau1984 for the merge as always. Thank you.

openedx-webhooks commented:
@Cup0fCoffee 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/31159&entry.1484655318=https://github.com/Cup0fCoffee&entry.752974735=2022-10-15+10:47&entry.1917517419=2022-10-20+14:01&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
